### DISCUSSION TITLE
About TSStandardize()

### DISCUSSION THREAD

**Discussion by Sure20220604** [timeseriesAI/tsai#543] [Q&A]
Hey,I want to consult that if I add TSStandardize() into clf:
![image](https://user-images.githubusercontent.com/106857313/173412991-6f959f01-0707-4091-acc6-cfd12d786ad8.png)
Then when I predicting ,do I need to Standardize my data again? Or I just put it in clf  and the clf help me do the Standardize atuomaticly. 
Sorry I'm new to this,if anyone can help me then I would really appreciate it.

**Comment by oguiza** [comment]
Hi @Sure20220604, 
The dataloader will automatically standardize the data, so you don’t need to add it a 2nd time.

**Reply by Sure20220604** [reply]
Hi @oguiza, I did a little test myself,I add  TSStandardize() into clf and train:
![image](https://user-images.githubusercontent.com/106857313/173593464-0fe4958e-8aa7-4e19-bc4f-64518c963cb2.png)
The the valid set's confusion matrix shows:
![image](https://user-images.githubusercontent.com/106857313/173593772-6644fb80-5bbc-4a31-89f6-ac7a0a6241f2.png)
I use valid set data to test it:
![image](https://user-images.githubusercontent.com/106857313/173594428-862f9f7d-e726-4cd6-9c58-98e6ae3deeeb.png)
Only when I use Standardize() into valid set(XXsplits[1]) and put data into clf.get_X_preds,my confusion matrix is the same with before,if I just use clf.get_X_preds(XXsplits[1]) ,then it's not 100%,if  the data is standardized automatically, then why is that?

**Comment by Sure20220604** [comment]
![image](https://user-images.githubusercontent.com/106857313/173671785-e4dd30c3-1cb9-422e-b34a-f2baaf784209.png)
![image](https://user-images.githubusercontent.com/106857313/173671838-97a22877-6ff7-4fe2-a3a9-c357cf37cfc7.png)
I use the tsai's sample to check,then it can match your sayings,so I'm a little confused.

**Reply by vrodriguezf** [reply]
Bear in mind that the TSStandardize transform of the dataloaders is normalising the data using the mean and std of the training set (as it should be) whereas if you do it manually with sklearn you're normalising with the mean and std of the validation set instead

**Reply by Sure20220604** [reply]
Okay,But if I don't have to do anything with my data anymore if I add TSStandarize(),why these result are different?(machine told me valid set accuracy is 100%;but I use clf.get_X_preds()shows only 50%):
![image](https://user-images.githubusercontent.com/106857313/174474297-176d6acc-f0fb-430c-ae9b-5627aa90ec6c.png)
![image](https://user-images.githubusercontent.com/106857313/174474305-aab4755f-65be-4b83-84cc-4d9de389e018.png)
![image](https://user-images.githubusercontent.com/106857313/174474342-0015a577-2487-492a-84a9-837c37ec1e96.png)

**Reply by zmce2018** [reply]
Hi @Sure20220604, 

If don't have to do anything with the data, you set batch_tfms = None.
